### PR TITLE
HW2

### PR DESCRIPTION
**Pull request by MaxSihua**

### CODE REVIEW

**Comment by KryvDanylo** [comment]
@MaxSihua прибери будь-ласка node_modules з коміту завдяки .gitignore файлу

File: index.ts
Comment (KryvDanylo) [inline index.ts:70]: Зроби реалізацію усіх роутів в окремій папці routes

File: .eslintrc.js
Comment (IhorOsadchyi) [inline .eslintrc.js:1]: You have similar content files in .js and json format for eslint configuration. One of the files seems redundant.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: It is more convenient to have interfaces and types in a separate file. As long as the project is small, you can start with the types.ts file, and then expand or divide it as needed.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: There is already import from express, so you can merge these lines.

```suggestion
import express, { Application, Request, Response } from 'express';
import swaggerJsdoc, { OAS3Definition } from 'swagger-jsdoc';
import swaggerUi from 'swagger-ui-express';
```

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: As `index.ts` file grows, may be better to start splitting things and move to it own separate modules. For example, swagger related configs can be move to `swagger.ts` module and then imported here.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: Similarly for resources. It is easier to maintain the project when resources and related files are in separate modules in subfolders.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: The same as above.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: And this too.

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: .eslintrc.js
Comment (MaxSihua) [reply .eslintrc.js:1]: done

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: done

File: src/assets/interface.ts
Comment (IhorOsadchyi) [inline src/assets/interface.ts:1]: I think `assets` is not the best place for this file. `Assets` folder commonly used for files like images or other media files.

Comment (MaxSihua) [reply src/assets/interface.ts:1]: done

File: src/swagger/resources.ts
Comment (IhorOsadchyi) [inline src/swagger/resources.ts:1]: You understood me not correct way. These annotations should be before the routes they describe, but route could be moved to separate file.

For example `healthCheck.ts`
```
import express, { Request, Response } from 'express';

interface ServerSatatus {
    status: string;
    message: string;
  }

const router = express.Router();
const serverIsWorking: ServerSatatus = {
    status: 'ok',
    message: 'Server is working!',
  };

/**
 * @swagger
 * /health-check:
 *   get:
 *     description: Check if the server is up and running
 *     responses:
 *       200:
 *         description: Server is up and running
 */
router.get('/', (req: Request, res: Response): void => {
  res.json(serverIsWorking);
});

export default router;
```

and in index.ts use just import and one line of code 

```
import healthCheckRouter from './resources/healthCheck'
app.use('/health-check', healthCheckRouter)
```

Comment (IhorOsadchyi) [reply src/swagger/resources.ts:1]: Updated comment

Comment (MaxSihua) [reply src/swagger/resources.ts:1]: I hope I understood correctly this time. Done.

File: src/index.ts
Comment (IhorOsadchyi) [inline src/index.ts]: Lets move this variable to separate `config.ts` file because it also used in swagger module.

```suggestion
import { port } from './config';
```

`config.ts`:
```
export const port = 3000;
```

File: package.json
Comment (IhorOsadchyi) [inline package.json]: This dependency should be in devDependencies section. As `nodemon` too. All dependenciess that used just for development should be moved to dev section.

Comment (IhorOsadchyi) [reply package.json]: Also, I found that `ts-node` dependency is missing in dev dependencies. Probably, it installed globally on you machine if it works without it. Better to have all required dependencies in local project's package.json file.

```
npm i -D nodemon @types/express ts-node
```

File: src/swagger/resources.ts
Comment (IhorOsadchyi) [reply src/swagger/resources.ts:1]: Almost all fine but small fixes could be applied. See additional comments.

File: src/index.ts
Comment (MaxSihua) [reply src/index.ts]: Done

File: package.json
Comment (MaxSihua) [reply package.json]: Done

File: src/swagger/configs.ts
Comment (IhorOsadchyi) [inline src/swagger/configs.ts]: Better to move it to separate `configs.ts` file in `src` folder not related to swagger and import here.

**Comment by ihor-osadchyi-dev** [comment]
Fix last comment and merge.

**Review by IhorOsadchyi — APPROVED** [review]

**Review by IhorOsadchyi — APPROVED** [review]
Fix last comment and merge.

**Comment by MaxSihua** [comment]
> Виправити останній коментар і об’єднати.

I've made corrections and merged. If everything is okay, I'm waiting for approval on the learn platform.

The third task is also ready, but now I'm going to make all the same corrections as in the second task, and then I'll create a Pull Request.

**Pull request merged**
